backend/index.js

```diff
@@ -0,0 +1,68 @@
+const express = require('express')
+const app = express()
+app.use(express.json())
+const mongoose = require('mongoose')
+const connection_url = "mongodb+srv://newuser:newPassword@cluster0.w4wyn.mongodb.net/RecipeDB?retryWrites=true&w=majority"
+
+mongoose.connect(connection_url)
+.then(()=> console.log('Succesfully Connected'))
+.catch((error)=> console.log('Could not connect due to ${error}'))
+
+const Recipe = require("./models/recipeSchema")
+
+app.get('/api/recipe', async (req, res) => {
+  const recipe = await Recipe.find({})
+  res.send(recipe)
+})
+
+app.get('/api/recipes/:recipeName', async (req, res) => {
+  const recipeName = await Recipes.findOne({recipeName: req.params.recipeName})
+  res.send(recipeName)
+})
+
+app.post('/api/recipe', async (req, res) => {
+  const {id, recipeName, link, recipeDescription, recipeImage, ingredientList, steps} = req.body
+  let recipe = new Recipe({
+    id,
+    recipeName,
+    link,
+    recipeDescription,
+    recipeImage,
+    ingredientList,
+    steps
+  })
+  try{
+    recipe = await recipe.save()
+    res.send('Recipe named ${recipeName} added to collection.')
+  } catch(error){
+    res.status(500).send(error.message)
```

Comment:
rather than sending a generic error code of 500, we can put error.code here and it should display a more precise error message!